diff --git a/index.d.ts b/index.d.ts
index 2d617bcb6e6d378d99a89fdf37fd632eb9fa325b..094587a884e2dc28be3f2cfb77bb1089dd64e476 100755
--- a/index.d.ts
+++ b/index.d.ts
@@ -236,7 +236,7 @@ declare namespace React {
      * @deprecated Use either `ReactNode[]` if you need an array or `Iterable<ReactNode>` if its passed to a host component.
      */
     interface ReactNodeArray extends ReadonlyArray<ReactNode> {}
-    type ReactFragment = {} | Iterable<ReactNode>;
+    type ReactFragment = Iterable<ReactNode>;
     type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;
 
     //
