diff --git a/assets/stylesheets/_components.scss b/assets/stylesheets/_components.scss
index f89b5b9a4b..e40c68049e 100644
--- a/assets/stylesheets/_components.scss
+++ b/assets/stylesheets/_components.scss
@@ -107,6 +107,7 @@
 @import 'components/credit-card-form-fields/style';
 @import 'components/date-picker/style';
 @import 'components/dialog/style';
+@import 'components/diff-viewer/style';
 @import 'components/domains/domain-product-price/style';
 @import 'components/domains/domain-registration-suggestion/style';
 @import 'components/domains/domain-search-results/style';
diff --git a/client/components/diff-viewer/README.md b/client/components/diff-viewer/README.md
index e69de29bb2..f83b3ef4db 100644
--- a/client/components/diff-viewer/README.md
+++ b/client/components/diff-viewer/README.md
@@ -0,0 +1 @@
+# diff viewer
diff --git a/client/components/diff-viewer/docs/example.jsx b/client/components/diff-viewer/docs/example.jsx
index e69de29bb2..8616eff03e 100644
--- a/client/components/diff-viewer/docs/example.jsx
+++ b/client/components/diff-viewer/docs/example.jsx
@@ -0,0 +1,23 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import DiffViewer from '../index';
+import diff from './diff.json';
+
+DiffViewer.displayName = 'DiffViewer';
+
+const DiffViewerExample = ( { exampleCode } ) => exampleCode;
+
+DiffViewerExample.displayName = 'DiffViewer';
+
+DiffViewerExample.defaultProps = {
+	exampleCode: `<DiffViewer diff={ \`${ diff }\` />`,
+};
+
+export default DiffViewerExample;
diff --git a/client/components/diff-viewer/index.jsx b/client/components/diff-viewer/index.jsx
index e69de29bb2..ac0e43439f 100644
--- a/client/components/diff-viewer/index.jsx
+++ b/client/components/diff-viewer/index.jsx
@@ -0,0 +1,68 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Fragment } from 'react';
+import { parsePatch } from 'diff';
+
+const filename = ( { oldFileName, newFileName } ) => {
+	const oldName = oldFileName.replace( /.+\//, '' );
+	const newName = newFileName.replace( /.+\//, '' );
+
+	return oldName !== newName ? `${ oldName } → ${ newName }` : oldName;
+};
+
+export const DiffViewer = ( { diff } ) => (
+	<div className="diff-viewer">
+		{ parsePatch( diff ).map( ( file, fileIndex ) => (
+			<Fragment key={ fileIndex }>
+				<div key={ `file-${ fileIndex }` } className="diff-viewer__filename">
+					{ filename( file ) }
+				</div>
+				<div key={ `diff-${ fileIndex }` } className="diff-viewer__file">
+					<div key="left-numbers" className="diff-viewer__line-numbers">
+						{ file.hunks.map( ( hunk, hunkIndex ) => {
+							let lineOffset = 0;
+							return hunk.lines.map( ( line, index ) => (
+								<div key={ `${ hunkIndex }-${ index }` }>
+									{ line[ 0 ] === '+' ? '\u00a0' : hunk.oldStart + lineOffset++ }
+								</div>
+							) );
+						} ) }
+					</div>
+					<div key="right-numbers" className="diff-viewer__line-numbers">
+						{ file.hunks.map( ( hunk, hunkIndex ) => {
+							let lineOffset = 0;
+							return hunk.lines.map( ( line, index ) => (
+								<div key={ `${ hunkIndex }-${ index }` }>
+									{ line[ 0 ] === '-' ? '\u00a0' : hunk.newStart + lineOffset++ }
+								</div>
+							) );
+						} ) }
+					</div>
+					<div className="diff-viewer__lines">
+						{ file.hunks.map( ( hunk, hunkIndex ) =>
+							hunk.lines.map( ( line, index ) => {
+								const output = line.slice( 1 ).replace( /^\s*$/, '\u00a0' );
+								const key = `${ hunkIndex }-${ index }`;
+
+								switch ( line[ 0 ] ) {
+									case ' ':
+										return <div key={ key }>{ output }</div>;
+
+									case '-':
+										return <del key={ key }>{ output }</del>;
+
+									case '+':
+										return <ins key={ key }>{ output }</ins>;
+								}
+							} )
+						) }
+					</div>
+				</div>
+			</Fragment>
+		) ) }
+	</div>
+);
+
+export default DiffViewer;
diff --git a/client/components/diff-viewer/style.scss b/client/components/diff-viewer/style.scss
index e69de29bb2..52117d589f 100644
--- a/client/components/diff-viewer/style.scss
+++ b/client/components/diff-viewer/style.scss
@@ -0,0 +1,47 @@
+/** @format */
+
+.diff-viewer__filename {
+	padding: 2px 4px;
+	background-color: $gray-lighten-20;
+	font-weight: 500;
+}
+
+.diff-viewer__file {
+	background-color: $gray-lighten-30;
+	display: flex;
+	font-family: $monospace;
+	flex-direction: row;
+	overflow-x: scroll;
+	white-space: pre;
+}
+
+.diff-viewer__line-numbers {
+	padding: 0 8px;
+	display: flex;
+	flex-direction: column;
+	text-align: right;
+	background-color: $gray-lighten-20;
+	color: $gray-text-min;
+}
+
+.diff-viewer__lines {
+	display: flex;
+	flex-direction: column;
+	flex-grow: 1;
+	overflow-x: visible;
+
+	& div,
+	& del,
+	& ins {
+		padding: 0 8px;
+	}
+
+	& del {
+		background-color: lighten($alert-red, 20%);
+		text-decoration: none;
+	}
+
+	& ins {
+		background-color: lighten($alert-green, 20%);
+	}
+}
diff --git a/client/devdocs/design/component-examples.js b/client/devdocs/design/component-examples.js
index 257aab8a47..ef5fd797a1 100644
--- a/client/devdocs/design/component-examples.js
+++ b/client/devdocs/design/component-examples.js
@@ -20,6 +20,7 @@ export Collection from 'devdocs/design/search-collection';
 export Count from 'components/count/docs/example';
 export CountedTextareas from 'components/forms/counted-textarea/docs/example';
 export DatePicker from 'components/date-picker/docs/example';
+export DiffViewerExample from 'components/diff-viewer/docs/example';
 export DropZones from 'components/drop-zone/docs/example';
 export EllipsisMenu from 'components/ellipsis-menu/docs/example';
 export EmbedDialog from 'components/tinymce/plugins/embed/docs/example';
diff --git a/client/devdocs/design/index.jsx b/client/devdocs/design/index.jsx
index 9d365688b3..9a60def888 100644
--- a/client/devdocs/design/index.jsx
+++ b/client/devdocs/design/index.jsx
@@ -43,6 +43,7 @@ import Collection from 'devdocs/design/search-collection';
 import Count from 'components/count/docs/example';
 import CountedTextareas from 'components/forms/counted-textarea/docs/example';
 import DatePicker from 'components/date-picker/docs/example';
+import DiffViewerExample from 'components/diff-viewer/docs/example';
 import DropZones from 'components/drop-zone/docs/example';
 import EllipsisMenu from 'components/ellipsis-menu/docs/example';
 import EmbedDialog from 'components/tinymce/plugins/embed/docs/example';
@@ -172,6 +173,7 @@ class DesignAssets extends React.Component {
 					<Count readmeFilePath="count" />
 					<CountedTextareas readmeFilePath="forms/counted-textarea" />
 					<DatePicker readmeFilePath="date-picker" />
+					<DiffViewerExample readmeFilePath="diff-viewer" />
 					<DropZones searchKeywords="drag" readmeFilePath="drop-zone" />
 					<EllipsisMenu readmeFilePath="ellipsis-menu" />
 					<EmbedDialog readmeFilePath="tinymce/plugins/embed" />
diff --git a/client/devdocs/design/playground-scope.js b/client/devdocs/design/playground-scope.js
index 5afdbb1f13..62fd729889 100644
--- a/client/devdocs/design/playground-scope.js
+++ b/client/devdocs/design/playground-scope.js
@@ -27,6 +27,7 @@ export Collection from 'devdocs/design/search-collection';
 export Count from 'components/count';
 export CountedTextarea from 'components/forms/counted-textarea';
 export DatePicker from 'components/date-picker';
+export DiffViewer from 'components/diff-viewer';
 export DropZones from 'components/drop-zone';
 export EllipsisMenu from 'components/ellipsis-menu';
 export Emojify from 'components/emojify';
diff --git a/client/my-sites/stats/activity-log/threat-alert.jsx b/client/my-sites/stats/activity-log/threat-alert.jsx
index 1ada584c81..79dc5afa93 100644
--- a/client/my-sites/stats/activity-log/threat-alert.jsx
+++ b/client/my-sites/stats/activity-log/threat-alert.jsx
@@ -9,6 +9,7 @@ import { localize } from 'i18n-calypso';
  * Internal dependencies
  */
 import ActivityIcon from '../activity-log-item/activity-icon';
+import DiffViewer from 'components/diff-viewer';
 import FoldableCard from 'components/foldable-card';
 import TimeSince from 'components/time-since';
 
@@ -61,16 +62,7 @@ export class ThreatAlert extends Component {
 								/>
 							</div>
 							<p className="activity-log__threat-alert-dirpath">{ file.dirpath }</p>
-							{ file.diff && (
-								<Fragment>
-									<pre>
-										<code>{ file.diff[ 0 ] }</code>
-									</pre>
-									<pre>
-										<code>{ file.diff[ 1 ] }</code>
-									</pre>
-								</Fragment>
-							) }
+							{ file.diff && <DiffViewer diff={ file.diff } /> }
 						</Fragment>
 					) ) }
 			</FoldableCard>
